meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc

```diff
@@ -54,14 +54,11 @@ do_provide_mender_defines() {
     fi
 
     if [ -n "${MENDER_BOOT_PART}" ]; then
-        MENDER_BOOT_PART_NUMBER=`get_part_number_from_device ${MENDER_BOOT_PART}`
-        MENDER_BOOT_PART_NUMBER_HEX=`get_part_number_hex_from_device ${MENDER_BOOT_PART}`
+        MENDER_BOOT_PART_NUMBER_HEX=`number_to_hex ${MENDER_BOOT_PART_NUMBER}`
     fi
 
-    MENDER_ROOTFS_PART_A_NUMBER=`get_part_number_from_device ${MENDER_ROOTFS_PART_A}`
-    MENDER_ROOTFS_PART_A_NUMBER_HEX=`get_part_number_hex_from_device ${MENDER_ROOTFS_PART_A}`
-    MENDER_ROOTFS_PART_B_NUMBER=`get_part_number_from_device ${MENDER_ROOTFS_PART_B}`
-    MENDER_ROOTFS_PART_B_NUMBER_HEX=`get_part_number_hex_from_device ${MENDER_ROOTFS_PART_B}`
+    MENDER_ROOTFS_PART_A_NUMBER_HEX=`number_to_hex ${MENDER_ROOTFS_PART_A_NUMBER}`
+    MENDER_ROOTFS_PART_B_NUMBER_HEX=`number_to_hex ${MENDER_ROOTFS_PART_B_NUMBER}`
```

Comment:
`MENDER_ROOTFS_PART_A_NUMBER` is referred to below, so we need to keep it around. I suggest logic similar to this:

```
if ${@bb.utils.contains('DISTRO_FEATURES', 'mender-partuuid', 'true', 'false', d)}; then
    MENDER_ROOTFS_PART_A_NUMBER_HEX=`number_to_hex ${MENDER_ROOTFS_PART_A_NUMBER}`
    MENDER_ROOTFS_PART_B_NUMBER_HEX=`number_to_hex ${MENDER_ROOTFS_PART_B_NUMBER}`
else
    MENDER_ROOTFS_PART_A_NUMBER=`get_part_number_from_device ${MENDER_ROOTFS_PART_A}`
    MENDER_ROOTFS_PART_A_NUMBER_HEX=`get_part_number_hex_from_device ${MENDER_ROOTFS_PART_A}`
    MENDER_ROOTFS_PART_B_NUMBER=`get_part_number_from_device ${MENDER_ROOTFS_PART_B}`
    MENDER_ROOTFS_PART_B_NUMBER_HEX=`get_part_number_hex_from_device ${MENDER_ROOTFS_PART_B}`
fi
```

And then `get_part_number_from_device` can contain some detection code for UUIDs, and suggest enabling the `mender-partuuid` and setting the `MENDER_ROOTFS_PART_A_NUMBER` variables, in the error message.